Styling topics update for XD (#529)
IgniteUI/indigo-design-docfx: en/style/colors.md

---
title: Colors - Design System Styling
_description: The color styles let you set up palettes and special colors as the basis of a theme for the Indigo.Design System.
_keywords: Design Systems, Design Systems UX, UI kit, Sketch, Ignite UI for Angular, Sketch to Angular, Sketch to Angular, Angular, Angular Design System, Export code from Sketch, Design Kits for Angular, Sketch HTML, Sketch to HTML, Sketch UI kits
---

# Colors

> There are five palettes with 10 color variations each for the `primary`, `secondary`, `grays`, and two chart `sereis` colors, as well as special colors such as `success`, `warn`, `error`, `info`, and `surface` in the the **Indigo.Design System**. This lets you change the complete look of the library and all projects that use it with just a few clicks. The way Colors are set up is identical to the [Ignite UI for Angular Themes](https://www.infragistics.com/products/ignite-ui-angular/angular/components/themes.html).

Reviewer: ```suggestion
There are five palettes with 10 color variations each for the `primary`, `secondary`, `grays`, and two charts `series` colors, as well as special colors such as `success`, `warn`, `error`, `info`, and `surface` in the **Indigo.Design System**. This lets you change the complete look of the library and all projects that use it with just a few clicks. The way Colors are set up is identical to the [Ignite UI for Angular Themes](https://www.infragistics.com/products/ignite-ui-angular/angular/components/themes.html).
```

<img class="responsive-img" src="../images/colors_palette.png" />

In Sketch, these can be found on a separate page named `🎨 Colors` and are available as `Layer Styles` that can be applied to any shape element. In Adobe XD they are placed on an Artboard with the same name in the left column of the library file and exist as `Colors` that are used across components and patterns.

> The **Indigo.Design System** features special color palettes optimized for various data visualization scenarios. Both ensure good contrast between adjacent series colors so that the palette can be used e.g. to colorize ranges of a gauge. One of the palettes is further optimized to be 100% inclusive and facilitate easy comprehension by people with color vision defficiencies. 

Reviewer: ```suggestion
The **Indigo.Design System** features special color palettes optimized for various data visualization scenarios. Both ensure good contrast between adjacent series colors so that the palette can be used e.g. to colorize ranges of a gauge. One of the palettes is further optimized to be 100% inclusive and facilitate easy comprehension by people with color vision deficiencies. 
```

The same procedure can be followed for the `secondary` colors and other colors with special meaning in the library.

## Palette Generation in Adobe XD

> To change the primary palette in Adobe XD open the Indigo.Design library file and show the `Libraries` panel. Then, right-click on `primary.500` and choose `Edit` from the contextual menu that appears to specify a different color. You will notice that this will update the palette on the `🎨 Colors` artboard but only the `primary.500` color in the `Libraries` panel will reflect the change.

Reviewer: ```suggestion
To change the primary palette in Adobe XD open the Indigo.Design library file and show the `Libraries` panel. Then, right-click on `primary.500` and choose `Edit` from the contextual menu that appears to specify a different color. You will notice that this will update the palette on the `🎨 Colors` artboard, but only the `primary.500` color in the `Libraries` panel will reflect the change.
```

<img class="responsive-img" src="../images/colors-palette-xd1.png" srcset="../images/colors-palette-xd1@2x.png 2x"/>

> The rest of the colors in the `Libraries` panel have to be updated manually. To achieve this draw a rectangle and use the eyedropper tool to set its fill by picking a color variant from the palette on the `🎨 Colors` artboard. Then use the hex value to edit the respective color in the `Libraries` panel. Follow this process until the whole palette is updated.

Reviewer: ```suggestion
The rest of the colors in the `Libraries` panel have to be updated manually. To achieve this, draw a rectangle and use the eyedropper tool to set its fill by picking a color variant from the palette on the `🎨 Colors` artboard. Then use the hex value to edit the respective color in the `Libraries` panel. Follow this process until the whole palette is updated.
```

5.  You are all set, your custom color will appear as a Color Variable and show up in the overrides for your Components and Patterns.

## Creating Custom Colors in Adobe XD

There are cases where we need more colors than the ones we have available. There are two approaches for creating custom colors that differ in terms of the scope we do this for.

If you want your added colors to be available across all the projects that use the library, navigate to the `Colors` artboard in it, and zoom to the empty `Custom` section underneath the `series` palettes.

<img class="responsive-img" src="../images/colors_custom6.png" srcset="../images/colors_custom6@2x.png 2x"/>

> Then complete the steps below to finish the process. If you want your added colors to be accessible only within the scope of your current project follow the steps below from anywhere in it.

Reviewer: ```suggestion
Then complete the steps below to finish the process. If you want your added colors to be accessible only within the scope of your current project, follow the steps below from anywhere in it.
```

1.  Draw a rectangle and set its fill to the color of your choice. 

    <img class="responsive-img" src="../images/colors_custom7.png" srcset="../images/colors_custom7@2x.png 2x"/>

> 2.  Then, in the `Libraries` panel click on the `+` icon to the right of `Colors` to create the style and give it an appropriate name.

Reviewer: ```suggestion
2.  Then, in the `Libraries` panel, click on the `+` icon to the right of `Colors` to create the style and give it an appropriate name.
```
